xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -300,9 +300,10 @@ bool CGUIWindowPVRBase::InitChannelGroup()
     if (m_channelGroup != group)
     {
       m_viewControl.SetSelectedItem(0);
-      m_channelGroup = group;
-      m_vecItems->SetPath(GetDirectoryPath());
+      SetChannelGroup(group, false);
```

Comment:
> that both this method and SetChannelGroup() don't attempt to grab the same lock

 they do, but whats the problem with this? performance? no. these methods are not getting called very often.